MCprep_addon/spawner/mcmodel.py

```
@@ -277,12 +277,10 @@ def get_final_texture_key(texture_ref: str, textures: Dict[str, str], visited: O
 
 	# Avoid infinite recursion
 	if key in visited:
-		line, file = env.current_line_and_file()
-		return MCprepError(
-			ValueError("Circular texture reference"),
-			line,
-			file,
-			f"Circular texture reference detected for key: {key}")
+		(line, file) = env.current_line_and_file()
+		error_collector.collect(MCprepError(Exception("Texture recursion"), line, file, f"Texture recursion on key '{key}' | Using '{key}'"))
+		return key
```

I'm not a fan of this, cause the caller of the function is not aware that this technically failed. Functions should return errors if something fails, and leave it up to the caller of the function to handle however they wish.

Regarding bmesh corruption: if a model has an error in it, in my view, it would be better to abort creating the model entirely, and let the user know, than to silently error in the end.

---

Also as an aside, the parenthesis around `line` and `file` is entirely unnecessary. `env.current_line_and_file` returns a tuple, and Python will already interpret `line, file = env.current_line_and_file()` as a tuple unpacking operation.

---

I placed parentheses around `line` and `file` since I saw them in the other files, which made it a tad easier on my eyes.
I’ll remove them before taking this PR out of draft.